src/consensus.rs

```diff
@@ -18,17 +18,17 @@ pub struct Consensus<T: Proposition> {
     pub decision: Option<Decision<T>>,
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Decision<T: Proposition> {
     pub votes: BTreeSet<SignedVote<T>>,
     pub proposals: BTreeMap<T, Signature>,
     pub faults: BTreeSet<Fault<T>>,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub enum VoteResponse<T: Proposition> {
     WaitingForMoreVotes,
     Broadcast(SignedVote<T>),
-    Decided(Decision<T>),
```

Comment:
We now recursively process all of votes that we cast, we can't have a `Decided` response because we may reach a decision after our vote is cast, but, we still need to broadcast our vote so that others decide.

So, instead, the caller needs to query consensus' `Consensus::decision` to see if we've decided.